### PR TITLE
Added support for logging

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-crt-sys/src/lib.rs [hunk 1]
```diff
@@ -77,6 +77,62 @@ extern "C" {
...
+    pub const AWS_CRT_LOG_INFO: _aws_crt_log_level = _aws_crt_log_level(4);
+}
+impl _aws_crt_log_level {
+    pub const AWS_CRT_LOG_DEBUG: _aws_crt_log_level = _aws_crt_log_level(5);
+}
+impl _aws_crt_log_level {
+    pub const AWS_CRT_LOG_TRACE: _aws_crt_log_level = _aws_crt_log_level(6);
+}
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
```
Comment (bretambrose) [inline aws-crt-sys/src/lib.rs:127]: Possibly worth adding an ordering trait here (<, >) as that's used for filtering at the C level.

File: aws-crt-sys/src/lib.rs [hunk 2]
```diff
@@ -77,6 +77,62 @@ extern "C" {
...
+pub struct _aws_crt_log_level(pub ::std::os::raw::c_uint);
+pub use self::_aws_crt_log_level as aws_crt_log_level;
+extern "C" {
+    pub fn aws_crt_log_set_level(log_level: aws_crt_log_level);
+}
+extern "C" {
+    pub fn aws_crt_log_message(level: aws_crt_log_level, message: *const u8, length: usize);
```
Comment (bretambrose) [inline aws-crt-sys/src/lib.rs:134]: some places we're using u8 * and in other places we're doing ::std::os::raw::c_char * but I can't tell if it's consistent or not.

File: src/crt.h [hunk 3]
```diff
@@ -11,6 +11,9 @@
 #include "api.h"
 /* clang-format on */
 
+enum aws_napi_log_subject { AWS_LS_CRT = 0x900 };
```
Comment (bretambrose) [inline src/crt.h]: what's this, esp. the value? (AWS_LOG_SUBJECT_BEGIN_RANGE, AWS_LOG_SUBJECT_END_RANGE)

File: aws-crt-sys/src/lib.rs (hunk 1 above)
Comment (justinboswell) [reply aws-crt-sys/src/lib.rs:127]: This is all generated

File: aws-crt-sys/src/lib.rs (hunk 2 above)
Comment (justinboswell) [reply aws-crt-sys/src/lib.rs:134]: This is all generated

File: src/crt.h (hunk 3 above)
Comment (justinboswell) [reply src/crt.h]: I copied this from node to get moving, as you can tell by its shitty name, forgot to go back and fix it.

**Pull request merged**